x-pack/filebeat/input/http_endpoint/handler.go

```diff
@@ -52,6 +52,11 @@ type handler struct {
 }
 
 func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Few webhook servers don't send in the Content-Type , hence defaulting to application/json if not present
+	if r.Header.Get("Content-Type") == "" {
```

Comment:
The only reason to modify the request is to make the validator pass. So why not disable content-type validation using configuration?

From the [docs](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-http_endpoint.html):

<img width="289" alt="Screenshot 2024-01-24 at 13 34 14" src="https://github.com/elastic/beats/assets/4565752/77a2dd4a-a720-44db-adbe-22e7dcebed26">